### ISSUE TITLE
Train-accuracy=nan, my parameter setting error?

### ISSUE THREAD

**Issue by ysh329** [apache/incubator-mxnet#5528]
- I'm training a DeepID model. 
- Datum are `/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-train.rec` and `/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-val.rec` from `http://data.dmlc.ml/data/caltech-256/`.
- Code is edited, based on `/mxnet/example/image-classification/train_cifar10.py`.

I suspect that there is a problem with the parameter settings (see **Key Code**), resulting in the division calculation is wrong.

Besides, I also have questions below:
1. What does `num_examples` refer in `parser.set_defaults` (see **Key Code**)? number of train examples or test ones?
2. What does `image_shape` refer in `parser.set_defaults` (see **Key Code**)? How do I know image shape of `rec` data format. (Data is from `http://data.dmlc.ml/data/caltech-256`)

## Environment info
Operating System:
Jetson TX1, Ubuntu16.04 LTS 64bit, 4G RAM, MXNet 0.94

## Problem Message
```Shell
(py-mxnet-SSD) yuanshuai@tegra-ubuntu:~/sdcard/code/mxnet_inference/deepid$ python train_caltech-256-60_resize90.py 
INFO:root:start with arguments Namespace(batch_size=128, benchmark=0, data_nthreads=4, data_train='/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-train.rec', data_val='/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-val.rec', disp_batches=20, gpus='0', image_shape='3,90,90', kv_store='device', load_epoch=None, lr=0.0005, lr_factor=0.1, lr_step_epochs='200,250', max_random_aspect_ratio=0, max_random_h=36, max_random_l=50, max_random_rotate_angle=0, max_random_s=50, max_random_scale=1, max_random_shear_ratio=0, min_random_scale=1, model_prefix='./deepid-caltech-256-90*90', mom=0.9, monitor=0, network=None, num_classes=256, num_epochs=1, num_examples=30000, num_layers=None, optimizer='sgd', pad_size=4, random_crop=1, random_mirror=1, rgb_mean='123.68,116.779,103.939', test_io=0, top_k=0, wd=0.0001)
[16:07:18] src/io/iter_image_recordio.cc:209: ImageRecordIOParser: /home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-train.rec, use 1 threads for decoding..
[16:07:20] src/io/iter_image_recordio.cc:209: ImageRecordIOParser: /home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-val.rec, use 1 threads for decoding..
INFO:root:Epoch[0] Batch [20]   Speed: 280.45 samples/sec       Train-accuracy=0.002976
INFO:root:Epoch[0] Batch [40]   Speed: 284.43 samples/sec       Train-accuracy=0.005078
INFO:root:Epoch[0] Batch [60]   Speed: 282.84 samples/sec       Train-accuracy=0.004297
INFO:root:Epoch[0] Batch [80]   Speed: 284.47 samples/sec       Train-accuracy=0.005078
INFO:root:Epoch[0] Batch [100]  Speed: 286.06 samples/sec       Train-accuracy=0.003125
INFO:root:Epoch[0] Batch [120]  Speed: 285.53 samples/sec       Train-accuracy=0.002734
INFO:root:Epoch[0] Train-accuracy=nan
INFO:root:Epoch[0] Time cost=55.883
INFO:root:Saved checkpoint to "./deepid-caltech-256-90*90-0001.params"
INFO:root:Epoch[0] Validation-accuracy=0.002495
```
## Key Code
```Python
    parser.set_defaults(
        # network
        #network        = 'resnet',
        #num_layers     = 110,
        # data
        data_train            = train_fname,
        data_val              = val_fname,
        num_classes       = 256,
        num_examples   = 30000,
        image_shape      = '3,90,90',#biggest:90
        pad_size             = 4,
        # train
        gpus           = '0',
        batch_size     = 128,
        num_epochs     = 1,#300
        lr             = .0005,#.05
        lr_step_epochs = '200,250',
        model_prefix   = './deepid-caltech-256-90*90'
    )
    args = parser.parse_args()

    # load network
    from importlib import import_module
    #net = import_module('symbols.'+args.network)
    #sym = net.get_symbol(**vars(args))
    sym = get_symbol(256)
```

## Complete Code
```Python
import os,sys
import argparse
import logging
logging.basicConfig(level=logging.DEBUG)
from common import find_mxnet, data, fit
from common.util import download_file
import mxnet as mx

sys.path.append('/mnt/sdcard/lizhihao/install/opencv-3.1.0/lib/python2.7/dist-packages')

def get_symbol(num_classes = 10):# default:530
    input_data = mx.symbol.Variable(name="data")
    # stage 1
    conv1 = mx.symbol.Convolution(
        data=input_data, kernel=(4, 4), stride=(1, 1), num_filter=20)
    relu1 = mx.symbol.Activation(data=conv1, act_type="relu")
    pool1 = mx.symbol.Pooling(
        data=relu1, pool_type="max", kernel=(2, 2), stride=(2,2))

    # stage 2
    conv2 = mx.symbol.Convolution(
        data=pool1, kernel=(3, 3), stride=(1, 1), num_filter=40)#40
    relu2 = mx.symbol.Activation(data=conv2, act_type="relu")
    pool2 = mx.symbol.Pooling(data=relu2, kernel=(2, 2), stride=(1, 1), pool_type="max")

    # stage 3
    conv3 = mx.symbol.Convolution(
        data=pool2, kernel=(3, 3), pad=(1, 1), num_filter=60)#60
    relu3 = mx.symbol.Activation(data=conv3, act_type="relu")
    pool3 = mx.symbol.Pooling(data=relu3, kernel=(2, 2), stride=(2, 2), pool_type="max")


    conv4 = mx.symbol.Convolution(data=pool3, kernel=(2, 2), pad=(1, 1), num_filter=80)#80
    relu4 = mx.symbol.Activation(data=conv4, act_type="relu")

    # stage 4
    fc160_1 = mx.symbol.FullyConnected(data=relu4, num_hidden=160)#160
    fc160_2 = mx.symbol.FullyConnected(data=conv3, num_hidden=160)#160

    dropout = mx.symbol.Dropout(data=fc160_2, p=0.5)
    fc_class = mx.symbol.FullyConnected(data=dropout, num_hidden=num_classes)
    # stage 6
    softmax = mx.symbol.SoftmaxOutput(data=fc_class, name='softmax')
    return softmax

def download_cifar10():
    data_dir="data"
    fnames = (os.path.join(data_dir, "cifar10_train.rec"),
              os.path.join(data_dir, "cifar10_val.rec"))
    #download_file('http://data.mxnet.io/data/cifar10/cifar10_val.rec', fnames[1])
    #download_file('http://data.mxnet.io/data/cifar10/cifar10_train.rec', fnames[0])
    #download_file('/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-val.rec', fnames[1])
    #download_file('/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-train.rec', fnames[0])
    fnames = list(fnames)
    fnames[0] = '/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-train.rec'
    fnames[1] = '/home/yuanshuai/sdcard/download/caltech-256/caltech-256-60-val.rec'
    return fnames

if __name__ == '__main__':
    # download data
    (train_fname, val_fname) = download_cifar10()

    # parse args
    parser = argparse.ArgumentParser(description="train cifar10",
                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    fit.add_fit_args(parser)
    data.add_data_args(parser)
    data.add_data_aug_args(parser)
    data.set_data_aug_level(parser, 2)
    parser.set_defaults(
        # network
        #network        = 'resnet',
        #num_layers     = 110,
        # data
        data_train     = train_fname,
        data_val       = val_fname,
        num_classes    = 256,
        num_examples  = 30000,
        image_shape    = '3,90,90',#biggest:90
        pad_size       = 4,
        # train
        gpus           = '0',
        batch_size     = 128,
        num_epochs     = 1,#300
        lr             = .0005,#.05
        lr_step_epochs = '200,250',
        model_prefix   = './deepid-caltech-256-90*90'
    )
    args = parser.parse_args()

    # load network
    from importlib import import_module
    #net = import_module('symbols.'+args.network)
    #sym = net.get_symbol(**vars(args))
    sym = get_symbol(256)

    # train
    fit.fit(args, sym, data.get_rec_iter)
```

**Comment by salemmohammed** [comment]
Hi @ysh329 ,

I am having the same problem with different setting. I report that on @5516. I hope somebody reply.

**Comment by mcxkarakoc** [comment]
.

**Comment by ChristianEschen** [comment]
I have same problem training a CNN.

I wonder if it has something to do with the metric function as proposed in https://github.com/dmlc/mxnet/issues/4253 ?

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!

**Comment by maxenceliu** [comment]
IF number of batches in one Epoch is integral multiple of display-frequent. Train-accuracy will display Nan all the time. 
Furthermore, there is no need to look train-accuracy after one epoch. Only validation-accuracy matters.

**Comment by indhub** [comment]
This should fix it: https://github.com/apache/incubator-mxnet/pull/10437
